### PR TITLE
new: fix connected textures crash

### PR DESCRIPTION
**Pull request by Wyvest**

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-bin.zip
```
Comment (asbyth) [inline gradle/wrapper/gradle-wrapper.properties:3]: Generally avoid unnecessary changes like these.

File: src/main/java/club/sk1er/patcher/mixins/bugfixes/optifine/RegionRenderCacheMixin_FixConnectedTextures.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    @Final
+    private static IBlockState DEFAULT_STATE;
+
+    @Shadow
+    protected abstract int getPositionIndex(BlockPos p_getPositionIndex_1_);
+
+    @Shadow
+    private IBlockState[] blockStates;
+
+    // yes, getPositionIndex is called twice because of this, but it doesn't really matter
```
Comment (asbyth) [inline src/main/java/club/sk1er/patcher/mixins/bugfixes/optifine/RegionRenderCacheMixin_FixConnectedTextures.java:27]: Rewrote the mixin so this doesn't need to be called twice, tested & works in prod.
https://hst.sh/diqojikayi.cpp

File: src/main/java/club/sk1er/patcher/mixins/bugfixes/optifine/RegionRenderCacheMixin_FixConnectedTextures.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import net.minecraft.client.renderer.RegionRenderCache;
+import net.minecraft.util.BlockPos;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@SuppressWarnings("UnusedMixin") //added in PatcherMixinPlugin
```
Comment (asbyth) [inline src/main/java/club/sk1er/patcher/mixins/bugfixes/optifine/RegionRenderCacheMixin_FixConnectedTextures.java:14]: Unnecessary, our MixinPlugin stops anything in a `.optifine` package from being applied if OptiFine isn't found.

File: src/main/java/club/sk1er/patcher/mixins/plugin/PatcherMixinPlugin.java [hunk 4]
```diff
@@ -52,7 +53,11 @@ public void acceptTargets(Set<String> myTargets, Set<String> otherTargets) {
 
     @Override
     public List<String> getMixins() {
-        return null;
+        //#if MC==10809
+        return Lists.newArrayList("bugfixes.optifine.RegionRenderCacheMixin_FixConnectedTextures");
```
Comment (asbyth) [inline src/main/java/club/sk1er/patcher/mixins/plugin/PatcherMixinPlugin.java:57]: Unnecessary, our MixinPlugin stops anything in a `.optifine` package from being applied if OptiFine isn't found.

File: src/main/resources/patcher.mixins.json [hunk 5]
```diff
@@ -7,7 +7,6 @@
     "conformVisibility": true
   },
   "client": [
-    "accessors.AbstractClientPlayerAccessor",
```
Comment (asbyth) [inline src/main/resources/patcher.mixins.json:10]: Generally avoid unnecessary changes like these & the RegionRenderCacheMixin needs to be in this file.

**Review by asbyth — CHANGES_REQUESTED** [review]

**Review by asbyth — APPROVED** [review]

**Review by LlamaLad7 — APPROVED** [review]
The Mixin itself looks fine, the fix itself seems like a bit of a band-aid but I can't immediately see a better one.
